v8pp/context.hpp

```diff
@@ -68,7 +70,42 @@ class context
 		return set(name, cl.js_function_template()->GetFunction(isolate_->GetCurrentContext()).ToLocalChecked());
 	}
 
+    /// Enter the context 
+    void enter();
+
+    /// Exit the context
+    void exit();
+
+    /// The context_scope class is an RAII guard to enter and exit the contex's scope. 
+    /// In Multithreaded scenarios the users of the library must enter the context before running.
+    /// Note, 
+    /// 1. If the context owns the Isolate then this guard will also Enter the isolate.
+    /// 2. If the context owns the Isolate then this will also aquire the Locker for this thread  
+    class context_scope {
+      public:
+        context_scope(context &context);
+        ~context_scope();
+        context_scope(context_scope &&other) = delete;
+        context_scope(const context_scope &other) = delete;
+        context_scope &operator=(const context_scope &other) = delete;
+
+      private:
+        context &context_;
+        struct empty_struct {};
+        struct v8locks {
+          v8::Locker locker;
+          v8::Isolate::Scope scope;
+		  v8::HandleScope handle;
+        };
+        bool owns_locks_;
+        union {
+          empty_struct empty_;
+          v8locks locks_;
+        };
+    };
+
```

Comment:
I've seen in one of your pull requests a `v8pp:context::impl()` function that returns a local handle to the `v8::Context`, encapsulated in the `v8pp::context` instance.

I would suggest to to add this `context::impl()` function in this PR. Having that we could use all other `v8::Context` methods, as well as the existing `v8::Context::Scope` that does scoped enter/exit calls. I think it is the very same as the proposed `context_scope`.

That `enter_context` parameter for the `v8pp::context` ctor seems great, I will add it. 

By the way, `v8pp :: context` ctor got a number of bool arguments, so it might be better to group them in a `v8pp :: context :: options` struct, until C++20 will come with designated initializers.